JS/Lesson34/goksel/recipe-app/src/modules/user/LogIn/index.tsx

```diff
@@ -0,0 +1,124 @@
+import Avatar from '@mui/material/Avatar';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import { Box, Button, Checkbox, Container, FormControl, FormControlLabel, FormHelperText, Grid, IconButton, InputAdornment, InputLabel, OutlinedInput, TextField, Typography } from '@mui/material';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useForm } from "react-hook-form";
+import { DataProps } from '../models';
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from 'react-redux';
+import { toggleSignIn } from '../userSlice';
+
+export const LogIn = () => {
+
+    const dispatch=useDispatch();
```

Comment:
You code would be cleaner if you always place hooks first in the component, then add functions (e.g. `const onSubmit = ...`), and in the end return statement with jsx code.

Reply:
Also, don't forget to format your files.

Reply:
I think ı fixed it.